### DISCUSSION TITLE
How to handle unsupported `kind` in generators?

### DISCUSSION THREAD

**Discussion by Jarod42** [premake/premake-core#1756] [Q&A]
I want to add support of `Utility`/`None` to some generators.

No sure of expected behavior you expected for unsupported `kind` (might also apply to other feature (`language` currently)).

We currently have warning for unsupported `kind` (fine), but problematic projects/configs are still passed to generators, which might produce hard errors :-/

Shouldn't we
- skip unsupported projects (from premake) (links or other dependencies might be broken)? (My second preferred solution).
- continue (as currently) ? (So generators has to handle as minima the unsupported project to avoid hard error).
- stop the whole generation? (As state above, chance are that generated solution would be broken somewhere). (My preferred solution)

Specifically for `kind None`:
For Generators as gmake2, which should do nothing for `None`,
should they state they support it and ignore the project? (My preferred solution)
or not supporting them and let premake ignore the project?

**Comment by starkos** [comment]
For gmake2, I agree that it should ignore `None` projects. Otherwise, I think the current behavior of warning on unsupported project kinds is the correct behavior.
